diff/diff_test.go

```diff
@@ -0,0 +1,96 @@
+package diff_test
+
+import (
+	"fmt"
+	"runtime/debug"
+	"strings"
+	"testing"
+
+	"github.com/poy/onpar"
+	"github.com/poy/onpar/diff"
+)
+
+type testStruct struct {
+	Foo string
+	Bar int
+
+	// unexported is not explicitly tested, but if our code attempts to compare
+	// unexported fields, it will panic.  This is here to ensure that we notice
+	// that.
+	unexported *testStruct
+}
+
+func TestDiff(t *testing.T) {
+	o := onpar.New()
+	defer o.Run(t)
+
+	for _, tt := range []struct {
+		name  string
+		value interface{}
+	}{
+		{"string", "this is a string"},
+		{"int", 21},
+		{"slice", []string{"this", "is", "a", "slice", "of", "strings"}},
+		{"map", map[string]string{"maps": "should", "work": "too"}},
+		{"struct", testStruct{Foo: "foo", Bar: 42}},
+		{"pointer", &testStruct{Foo: "foo", Bar: 42}},
+	} {
+		tt := tt
```

Comment:
Yep.  Without it, the only `tt` that is tested is `pointer` - because the loop reuses the memory for tt for each iteration of the loop, and the tests are running in separate goroutines.

I actually wanted to go back to see if I could clean that up some.  I don't know if there's an easy way to do that, though - maybe I should just switch them to running in serial?  Or I could name the struct type and set a BeforeEach somehow...

I'll do some thinking on it today.

Reply:
Ah, I see. It IS actually using it via closure. NVM.